src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -68,18 +68,23 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
 
         private void ActivateProperties(object controller, Type containerType, Dictionary<string, object> properties)
         {
+            if (properties.Count == 0)
+            {
+                return;
+            }
+
             var propertyHelpers = PropertyHelper.GetProperties(controller);
             foreach (var property in properties)
             {
-                var propertyHelper = propertyHelpers.First(helper => helper.Name == property.Key);
+                var propertyHelper = propertyHelpers.First(helper =>
+                    string.Equals(helper.Name, property.Key, StringComparison.Ordinal));
```

Comment:
should we throw a nice exception if no setter can be found? This might happen is someone sets up their own ADs.